trie/zk_trie_impl.go

```diff
@@ -665,21 +723,16 @@ func (mt *ZkTrieImpl) walk(nodeHash *zkt.Hash, f func(*Node)) error {
 	if err != nil {
 		return err
 	}
-	switch n.Type {
-	case NodeTypeEmpty:
+	if n.IsTerminal() {
 		f(n)
-	case NodeTypeLeaf:
-		f(n)
-	case NodeTypeParent:
+	} else {
 		f(n)
 		if err := mt.walk(n.ChildL, f); err != nil {
 			return err
 		}
 		if err := mt.walk(n.ChildR, f); err != nil {
 			return err
 		}
```

Comment:
A code style discussion: what if changing `IsTerminal` to `IsEmptyOrLeaf`, and moving other cases outside `IsTerminal`?